app/javascript/dashboard/components/ui/DatePicker/components/CalendarDateInput.vue

```diff
@@ -0,0 +1,76 @@
+<script setup>
+import { defineProps, defineEmits } from 'vue';
+import { parseDateFromDMY } from '../helpers/DatePickerHelper';
+
+import { isValid, startOfDay, endOfDay, isBefore, isAfter } from 'date-fns';
+
+const props = defineProps({
+  calendarType: {
+    type: String,
+    default: '',
+  },
+  dateValue: {
+    type: String,
+    default: '',
+  },
+  compareDate: {
+    type: String,
+    default: '',
+  },
+  isDisabled: Boolean,
+});
+
+const emit = defineEmits(['update', 'validate', 'error']);
+
+const updateDate = newValue => {
+  emit('update', newValue);
+};
+
+const validateDate = dateString => {
+  const parsedDate = parseDateFromDMY(dateString);
+
+  if (!isValid(parsedDate)) {
+    emit('error', 'Please select a valid time range');
+    return;
+  }
+
+  if (props.calendarType === 'start') {
+    if (
+      props.compareDate &&
+      isAfter(parsedDate, parseDateFromDMY(props.compareDate))
+    ) {
+      emit('error', 'Start date must be before end date');
+    } else {
+      emit('validate', startOfDay(parsedDate));
+    }
+  } else if (
+    props.compareDate &&
+    isBefore(parsedDate, parseDateFromDMY(props.compareDate))
+  ) {
+    emit('error', 'End date must be after start date');
+  } else {
+    emit('validate', endOfDay(parsedDate));
+  }
+};
+</script>
+
+<template>
+  <div class="h-[82px] flex flex-col items-start px-5 gap-1.5 pt-4 w-full">
+    <span class="text-sm font-medium text-slate-800 dark:text-slate-50">
+      {{
+        calendarType === 'start'
+          ? $t('DATE_PICKER.DATE_RANGE_INPUT.START')
+          : $t('DATE_PICKER.DATE_RANGE_INPUT.END')
+      }}
+    </span>
+    <input
+      type="text"
+      :value="dateValue"
+      class="reset-base border bg-slate-25 dark:bg-slate-900 border-slate-50 dark:border-slate-700/50 w-full disabled:text-slate-200 dark:disabled:text-slate-700 disabled:cursor-not-allowed text-slate-800 dark:text-slate-50 px-1.5 py-1 text-sm rounded-xl h-10"
+      placeholder="DD/MM/YYYY"
```

Comment:
We cannot use this format across the app. The way to handle this is quite straightforward, your components basically have two layers, a logic/computational layer and a presentation layer

```
Presentation - Dates showing up in the UI     
--------------------------------------------------------
Logic / Compute            
               - Toggling Views
               - Setting new values
               - Passing dates
               - Validating dates
```

The data inside the presentation layer can be of any format and can be formatted to the user locale, and the presentation layer is at the boundaries of the component (on the surface basically) Any operation within the component should use a standard data type across the board, the best is to use a `Date` object itself. The other option is to use the standard ISO date format, or a UNIX epoch

In this particular case, the value inside the input will be a computed value based on the `dateValue` prop and the user locale. While the prop that is passed is passed in a standard format. It's also emitted in events in a standard format